java/core/src/java/org/apache/orc/impl/StringHashTableDictionary.java

```diff
@@ -120,6 +122,11 @@ public void getText(Text result, int positionInKeyOffset) {
     DictionaryUtils.getTextInternal(result, positionInKeyOffset, this.keyOffsets, this.byteArray);
   }
 
+  @Override
+  public ByteBuffer getText(int positionInKeyOffset) {
+    return DictionaryUtils.getTextInternal(positionInKeyOffset, this.keyOffsets, this.byteArray);
+  }
```

Comment:
Is it safe to be exporting this ByteBuffer? If it is reused / overwritten in some way, will that affect any of the existing code.

Specifically, I'm wondering about the now returned `ByteBuffer` from `setByteBuffer`, as the data can be returned without any copying.

Reply:
Hey @kbendick,

> Is it safe to be exporting this ByteBuffer? If it is reused / overwritten in some way, will that affect any of the existing code.

I guess everyone has a different scale of what "safe" is.  If your concern is regarding the callers changing values in the returned ByteBuffer affected the data stored in the Dictionary, then yes, that is a possibility right now.  There are two ways to handle that:

1. Document this behavior and assume the caller will handle the data responsibly
2. Return a read-only `ByteBuffer` which will prevent callers from modifying the data

I'm not sure I understand the second part of your question.

Reply:
On the second part, if you are referring to the new value returned by `setByteBuffer`, there's no functionality change there.  I am simply returning the passed in value for convenience:

```java
ByteBuffer bb = ByteBuffer.allocate(16);
dictionary.setByteBuffer(bb, 0, 16);
System.out.println(bb);

// v.s.

ByteBuffer bb = ByteBuffer.allocate(16);
System.out.println(dictionary.setByteBuffer(bb, 0, 16));
```

Reply:
That was indeed what I was referring to. That answers my question. Thank you @belugabehr!